G1ANT.Addon.Images/Commands/WaitForImageCommand.cs

```diff
@@ -46,47 +45,40 @@ public class Arguments : CommandArguments
             public override TimeSpanStructure Timeout { get; set; }
 
             [Argument(Tooltip = "Name of a variable where the command's result will be stored")]
-            public VariableStructure Result { get; set; } = new VariableStructure("result");
-            
+            public VariableStructure Result { get; set; } = new VariableStructure("result");            
         }
-        public WaitForImageCommand(AbstractScripter scripter) : base(scripter)
-        { }
+
+        public WaitForImageCommand(AbstractScripter scripter) : base(scripter) { }
+
         public void Execute(Arguments arguments)
         {
             if (arguments.Threshold.Value < 0 || arguments.Threshold.Value > 1)
             {
                 throw new ArgumentOutOfRangeException("Threshold must be a value from 0 to 1.");
             }
 
-            using (Bitmap bitmap1 = Imaging.OpenImageFile(arguments.Image.Value, nameof(arguments.Image)))
+            var timeout = (int)arguments.Timeout.Value.TotalMilliseconds;
+            var start = Environment.TickCount;
+            var foundRectangle = Rectangle.Empty;
+
+            while (Math.Abs(Environment.TickCount - start) < timeout && Scripter.Stopped == false && foundRectangle == Rectangle.Empty)
             {
-                int timeout = (int)arguments.Timeout.Value.TotalMilliseconds;
-                long start = Environment.TickCount;
-                Rectangle foundRectangle = Rectangle.Empty;
-                while (Math.Abs(Environment.TickCount - start) < timeout && Scripter.Stopped == false && foundRectangle == Rectangle.Empty)
+                using (var template = arguments.Image.OpenImage())
+                using (var source = arguments.ScreenSearchArea.GetScreenshot(arguments.Relative))
```

Comment:
It doesnt look like optimal solution. After our discussion i get it why its like that, but i still suggest looking into it for couple more minutes. Maybe you will find better solution